### PR TITLE
Add model for current claimable balances

### PR DESCRIPTION
**Pull request by amishas157**
<details>
  <summary>PR Checklist</summary>

### PR Structure

- [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
- [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
      otherwise).
- [ ] This PR's title starts with the jira ticket associated with the PR.

### Thoroughness

- [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
- [ ] I've updated the docs and README with the added features, breaking changes, new instructions on how to use the repository.

### Release planning

- [ ] I've decided if this PR requires a new major/minor/patch version accordingly to
    [semver](https://semver.org/), and I've changed the name of the BRANCH to release/* , feature/* or patch/* .
</details>

### What

This PR adds current_state model for claimable balances

### Why

Check details in Jira

### Known limitations

None

### CODE REVIEW

File: models/marts/ledger_current_state/claimable_balances_current.sql [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            , row_number()
+                over (
+                    partition by cb.balance_id
+                    order by cb.closed_at desc
+                ) as rn
+        from {{ ref('stg_claimable_balances') }} as cb
+        {% if is_incremental() %}
+            -- limit the number of partitions fetched incrementally
+            where
+                cb.closed_at >= timestamp_sub(current_timestamp(), interval 30 day)
```
Comment (sydneynotthecity) [inline models/marts/ledger_current_state/claimable_balances_current.sql:40]: Is `claimable_balances` partitioned on `closed_at`? We should be limiting the data selected by partition key, so I would double check the table schema.

Comment (chowbao) [reply models/marts/ledger_current_state/claimable_balances_current.sql:40]: I think `stg_claimable_balances` is a view so it won't need it's own partition definition. It should use the same partitioning from `claimable_balances` (which is batch_run_date unfortunately)

File: models/marts/ledger_current_state/claimable_balances_current.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+          - not_null:
+              config:
+                where: batch_run_date > current_datetime - interval 2 day
+
+      - name: closed_at
+        description: '{{ doc("closed_at") }}'
+        tests:
+          - not_null:
+              config:
+                where: closed_at > current_timestamp - interval 2 day
```
Comment (chowbao) [inline models/marts/ledger_current_state/claimable_balances_current.yml:113]: Relates to https://github.com/stellar/stellar-dbt-public/pull/66/files#r1689817033
The test: where: <field> should be changed to `batch_run_date` to leverage the table's partitioning

File: models/marts/ledger_current_state/claimable_balances_current.sql (hunk 1 above)
Comment (amishas157) [reply models/marts/ledger_current_state/claimable_balances_current.sql]: Addressed in https://github.com/stellar/stellar-dbt-public/pull/66/commits/fad7bb31de3760302cfbb9cd64c2b077e71a8832

File: models/marts/ledger_current_state/claimable_balances_current.yml (hunk 2 above)
Comment (amishas157) [reply models/marts/ledger_current_state/claimable_balances_current.yml]: Addressed in https://github.com/stellar/stellar-dbt-public/pull/66/commits/fad7bb31de3760302cfbb9cd64c2b077e71a8832

File: models/marts/ledger_current_state/claimable_balances_current.sql [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            , row_number()
+                over (
+                    partition by cb.balance_id
+                    order by cb.closed_at desc
+                ) as rn
+        from {{ ref('stg_claimable_balances') }} as cb
+        {% if is_incremental() %}
+            -- limit the number of partitions fetched incrementally
+            where
+                cb.batch_run_date >= timestamp_sub(current_timestamp(), interval 7 day)
```
Comment (chowbao) [inline models/marts/ledger_current_state/claimable_balances_current.sql:40]: Does this cause a datetime vs timestamp error?

Comment (amishas157) [reply models/marts/ledger_current_state/claimable_balances_current.sql]: Good catch. Addressed in https://github.com/stellar/stellar-dbt-public/pull/66/commits/d7e567e22d1627c6feb682298733216bfd8c072f

**Review by chowbao — APPROVED** [review]

**Comment by chowbao** [comment]
LGTM. I forget if dbt does an automatic --full-refresh run of the table the first time it is run. Otherwise you might need to manually run a --full-refresh to get this in prod

**Comment by amishas157** [comment]
> I forget if dbt does an automatic --full-refresh run of the table the first time it is run. Otherwise you might need to manually run a --full-refresh to get this in prod

It will run a full refresh.  I did not have to do anything in test explicitly.

**Pull request merged**